### DISCUSSION TITLE
Children Annotation

### DISCUSSION THREAD

**Discussion by dezren39** [kdl-org/kdl#344] [Ideas]
Allow annotating the children block in the same way that values and identifiers can be annotated.

**Comment by tabatkins** [comment]
Again, please lead with a use-case explaining why this would be useful, rather than just asking for a feature with no additional details.

Just like with key annotations, I'm not sure I see what this can do. A node having some instruction for how to interpret its child list makes sense, but it can only have one child list, so that information can just be expressed with a property or attribute.

**Reply by dezren39** [reply]
for dsl purposes, if i could annotate children block and property blocks, i could contain my requirements on users entirely within annotations, allowing them to use any/all properties/values/children as they see fit for their data.

without children block and property block annotations, yes i can have my users express that with a property or attribute, but which one, and what will they use when they want to use that property but i've reserved it? maybe i'll reserve a prefix in addition to #? not sure.

all of that would be something i would need to teach to people in addition to "sometimes you need to provide hints with parenthesis or i won't know what you mean".

ideally they can just type documents without hints and i will automatically know, but when that doesn't work, they can hint on the exact thing that isn't working "right", maybe specifying a related/common type name or a specific internal id to force how it's handled.
